src/electionguard/facades/collections.c

```diff
@@ -0,0 +1,145 @@
+#include "electionguard/collections.h"
+
+typedef struct eg_linked_list_node_t {
+    char *key;
+    char *value;
+    struct eg_linked_list_node_t *previous;
+    struct eg_linked_list_node_t *next;
+} eg_linked_list_node_t;
+
+struct eg_linked_list_s {
+    uint64_t count;
+    eg_linked_list_node_t *head;
+    eg_linked_list_node_t *tail;
+};
+
+eg_electionguard_status_t eg_linked_list_new(eg_linked_list_t **out_handle)
```

Comment:
this linked list implementation is pretty naive because its current use cases do not require something complicated. These are designed to have less than 5-10 elements in them so the only way to find an element is to know its index position before hand or to enumerate through them and read the values. if this need changes we can add a find function but at that point we may need to implement something more verbose. this structure also has no key safety and it is up to the caller who is filling the structure to know if there are collisions.there is some protection against that because of how this structure is actually consumed - as an input across the C boundary for an unordered map in the ciphertext election context.